airflow/www/utils.py

```diff
@@ -808,6 +815,64 @@ def get_col_default(self, col_name: str) -> Any:
     filter_converter_class = AirflowFilterConverter
 
 
+class DagRunCustomSQLAInterface(CustomSQLAInterface):
+    """
+    Overwrite of custom delete and delete_all methods for speeding up
+    the DagRun deletion when DagRun has a lot of task instances.
+    The default cascade deletion was performing very slowly when task instances were more than 10k.
+
+    """
+
+    def delete(self, item: Model, raise_exception: bool = False) -> bool:
+        try:
+            self._delete_files(item)
```

Comment:
I wonder if it is possible to avoid calling this non-public function. Would this work as expected?

```python
def delete(self, ...):
    self.session.query(TaskInstance).where(TaskInstance.run_id == item.run_id).delete()
    return super().delete(...)
```

instead?

Reply:
It's a protected method, and it's okay to call it in the inherited classes. The main idea of this method is that it should delete all instances in one transaction and roll back if something is wrong.

Reply:
> It's a protected method

Citation needed

Reply:
Sorry, I don't understand what you mean

Reply:
Removed using of the protected method and added the super method call